extensions/README.md

```diff
@@ -0,0 +1,362 @@
+   user-provided struct. The fields of that struct will be treated as extension
+   fields to embedding structure. This corresponds to the map-extension sockets
+   in the spec.
+2. Some elements can contain values of one of pre-determined types. New types
+   can be provided by registering a factory function for those types using an
+   appropriate function. This (mostly) corresponds to the type-choice extension
+   sockets in the spec.
+3. A couple of type-choice sockets (`$tag-rel-type-choice`,
+   `$corim-role-type-choice` and `$comid-role-type-choice`) define what, in
+   effect, are extensible enums. They allow providing additional values, rather
+   than types. This implementation provides registration functions for new
+   values for those types.
+
+> [!NOTE]
+> The CoRIM spec includes some CDDL from the CoSWID spec that also features
+> extension sockets. This code base does include the CoSWID implementation, and
+> so does not cover those extension points.
+
+
+## Map Extensions
+
+Map extensions allow extending CoRIM maps with additional keys, effectively
+defining new fields for the corresponding structures. In the code base, these
+can be identified by the embedded `Extensions` struct. These are
+
+- `comid.Comid`
+- `comid.Entity`
+- `comid.FlagsMap`
+- `comid.Mval`
+- `comid.Triples`
+- `corim.Entity`
+- `corim.Signer`
+- `corim.UnsignedCorim`
+
+To extend the above types, you need to define a struct containing your
+extensions and pass a pointer to an instance of that struct to the
+`RegisterExtensions()` method of the corresponding instance of the type that is
+being extended. This should be done as early as possible, before any marshaling
+is performed. The
+
+These types can be extended in two ways: by adding additional fields, and by
+introducing additional constraints over existing fields.
+
+### Adding new fields
+
+To add a new fields, simply add them to your extensions struct, ensuring that
+the `cbor` and `json` tags on those fields are set corrects. As CoRIM spec
+mandates integer keys, you must use `keyasint` option for the `cbor` tag.
+
+To access the values of those fields, you can call the extended type instance's
+`Extensions.Get()` passing in the name of the field you want to access. The
+name can be either the Go struct's field name, the name specified in the `json`
+tag, or (a string containing) the integer specified in the `cbor` tag.
+
+`Get()` returns an `interface{}`. There are equivalent `GetInt()`,
+`GetString()`, etc. methods that  perform the required conversions, and return
+the value of the indicated type, along with possible errors. ("Must" versions
+of these also exists, e.g. `MustGetString()`, that do not return an error).
+
+You can also get the pointer your extensions instance itself by calling
+extended type instance's `GetExtensions()`. This returns an `interface{}`, so
+you will need to type assert in order to be able to access the fields directly.
+
+### Introducing additional constraints
```

Comment:
```suggestion
#### Introducing additional constraints
```

Reply:
no three is correct -- it should be under "Map Extensions", at the same level as "adding new fields" as those are the two things you can do with map extensions.